src/abstractdataframe/reshape.jl

```diff
@@ -355,12 +364,52 @@ julia> unstack(df, :variable, :value, fill=0)
 ─────┼──────────────────────
    1 │ 1           1      2
    2 │ 2           3      0
+
+julia> df = DataFrame(cols=["a", "a", "b"], values=[1, 2, 4])
+3×2 DataFrame
+ Row │ cols    values
+     │ String  Int64
+─────┼────────────────
+   1 │ a            1
+   2 │ a            2
+   3 │ b            4
+
+julia> unstack(df, :cols, :values, allowduplicates=identity)
+1×2 DataFrame
+ Row │ a       b
+     │ Array…  Array…
+─────┼────────────────
+   1 │ [1, 2]  [4]
+
+julia> unstack(df, :cols, :values, allowduplicates=sum)
+1×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     3      4
 ```
-Note that there are some differences between the widened results above.
 """
 function unstack(df::AbstractDataFrame, rowkeys, colkey::ColumnIndex,
                  value::ColumnIndex; renamecols::Function=identity,
-                 allowmissing::Bool=false, allowduplicates::Bool=false, fill=missing)
+                 allowmissing::Bool=false,
+                 allowduplicates::Union{Bool, Function}=false, fill=missing)
+    if allowduplicates isa Function
+        groupcols = vcat(index(df)[rowkeys], index(df)[colkey])
+        @assert groupcols isa AbstractVector{Int}
+        gdf = groupby(df, groupcols)
+        if check_aggregate(allowduplicates, df[!, value]) isa AbstractAggregate
```

Comment:
Add a comment to explain why this is needed?

Reply:
I have added the comment. The key issue is that `AbstractAggregate` is guaranteed to be a number, so we are sure we do not need to wrap it in `Ref` and then `combine` will use fast path.